### ISSUE TITLE
Getting "Cannot resolve path" error on all images

### ISSUE THREAD

**Issue by succli** [tcoopman/image-webpack-loader#71]
My webpack.config.js

```
var path = require('path')
var webpack = require('webpack')

const configs = [{
  entry: './assets/index.js',
  output: {
    path: path.resolve(__dirname, 'build'),
    filename: 'bundle.js'
  },
  module: {
    rules: [
      {
        test: /\.(js|jsx)$/,
        use: 'babel-loader'
      },
      {
        test: /\.(sass|scss)$/,
        use: [
          'style-loader',
          'css-loader',
          'sass-loader'
        ]
      },
      {
        test: /\.(png|jpg|gif|svg)$/,
        use: [
          'file-loader?context=assets/&name=[path][name].[ext]&publicPath=build/',
          {
            loader: 'image-webpack-loader',
            options: {
              mozjpeg: {
                progressive: true
              },
              gifsicle: {
                interlaced: false
              },
              optipng: {
                optimizationLevel: 4
              },
              pngquant: {
                quality: '75-90',
                speed: 3
              }
            }
          }
        ]
      }
    ]
  },

  plugins: [
    new webpack.ProvidePlugin({
      jQuery: 'jquery',
      $: 'jquery',
      jquery: 'jquery',
      'window.jQuery': 'jquery',
      Tether: 'tether'
    })
  ]
}
]

module.exports = configs
```

I get these errors after runing webpack:

```
Hash: bf148b3489e2aabad21e
Version: webpack 2.2.0
Child
    Hash: bf148b3489e2aabad21e
    Time: 14091ms
        Asset    Size  Chunks                    Chunk Names
    bundle.js  645 kB       0  [emitted]  [big]  main
      [72] ./assets/img/arrow-white-xl.png 562 bytes {0} [built] [failed] [1 error]
      [75] ./assets/js/app.js 680 bytes {0} [built]
      [76] ./assets/sass/app.scss 1.04 kB {0} [built]
      [77] ./assets/js/canvasmenu.js 2.15 kB {0} [built]
      [78] ./assets/js/homecarousel.js 3.56 kB {0} [built]
      [79] ./assets/js/map.js 3.34 kB {0} [built]
     [105] ./~/bootstrap/dist/js/bootstrap.js 106 kB {0} [built]
     [179] ./~/tether/dist/js/tether.js 60.4 kB {0} [built]
     [186] ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 57.7 kB {0} [built]
     [189] ./assets/img/arrow-black-down.png 562 bytes {0} [built] [failed] [1 error]
     [190] ./assets/img/arrow-gray-small.png 562 bytes {0} [built] [failed] [1 error]
     [191] ./assets/img/bg-silver-effect.png 562 bytes {0} [built] [failed] [1 error]
     [192] ./assets/img/home-carousel-box-sample-1.png 562 bytes {0} [built] [failed] [1 error]
     [201] ./~/style-loader/addStyles.js 7.15 kB {0} [built]
     [204] ./assets/index.js 63 bytes {0} [built]
        + 190 hidden modules

    ERROR in ./assets/img/pipe.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:2591-2617
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/arrow-black-down.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:1041-1079
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/arrow-white-xl.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:6594-6630 6:7380-7416
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/home-carousel-box-sample-1.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:10955-11003
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/logo-watermark.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:6069-6105
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/arrow-white.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:4962-4995 6:36814-36847 6:46891-46924
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/bg-silver-effect.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:5800-5838
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/arrow-gray-small.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:21027-21065
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/masonry-sample-1x1.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:16569-16609
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/masonry-sample-2x1.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:17496-17536
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/masonry-sample-1x2.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:18103-18143
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/reasons.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:8920-8949 6:9914-9943
     @ ./assets/sass/app.scss
     @ ./assets/index.js

    ERROR in ./assets/img/masonry-sample-2x2.png
    Module build failed: Error: A rendszer nem tal�lja a megadott el�r�si utat.

        at notFoundError (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:11:11)
        at verifyENOENT (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:46:16)
        at ChildProcess.cp.emit (F:\xampp\htdocs\hp-sitebuild\node_modules\execa\node_modules\cross-spawn\lib\enoent.js:33:19)
        at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
     @ ./~/css-loader!./~/sass-loader!./assets/sass/app.scss 6:18232-18272
     @ ./assets/sass/app.scss
     @ ./assets/index.js

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Evista\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
npm ERR! node v6.3.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! home-and-people@1.0.0 build: `npm rebuild node-sass && node node_modules/webpack/bin/webpack.js --optimize-minimize --config webpack.config.js`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the home-and-people@1.0.0 build script 'npm rebuild node-sass && node node_modules/webpack/bin/webpack.js --optimize-minimize --config webpack.config.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the home-and-people package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm rebuild node-sass && node node_modules/webpack/bin/webpack.js --optimize-minimize --config webpack.config.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs home-and-people
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls home-and-people
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     F:\xampp\htdocs\hp-sitebuild\npm-debug.log
```

"A rendszer nem találja a megadott elérési utat." means "The system cannot resolve/found defined path."

**Comment by tcoopman** [comment]
How does the require of the image look like and where are they located exactly.

**Comment by succli** [comment]
My directory view:
![devdir](https://cloud.githubusercontent.com/assets/9085766/23011810/406483b4-f423-11e6-9364-bd9a7be30d7a.png)

Requirement is from scss files (assets/sass/*.scss) like this: `background: url(../img/arrow-white.png) no-repeat center;`

**Comment by StarXY** [comment]
Hi, I have the same issue. when I comment this loader, The error doesn't exist. Here is the right configuration.
`
module.exports = () => {
    return {
        test: /\.(png|svg|jpg|gif)$/,
        use: [
            {
                loader: 'file-loader',
                options: {
                    name: '[name]-[hash:5].[ext]',
                    outputPath: '../assets/images/',
                    // 此处有bug， publicPath + outputPath == 图片路径？？？
                    publicPath: '../'
                    // 根据实际代码输出到相应位置
                    // useRelativePath: tru
                }
            },
            // {
            //     loader: 'image-webpack-loader',
            //     options: {
            //         mozjpeg: {
            //             quality: 65
            //         },
            //         pngquant:{
            //             quality: "65-90",
            //             speed: 4
            //         },
            //         svgo:{
            //             plugins: [
            //                 {
            //                     removeViewBox: false
            //                 },
            //                 {
            //                     removeEmptyAttrs: false
            //                 }
            //             ]
            //         }
            //     }
            // }
        ]
    };
};
`

**Comment by lostpebble** [comment]
> Here is the right configuration.

That is not really a solution.. because you are essentially not using this loader at all (no optimisation being done on your images).

I was running into this problem today. No matter how much I uninstalled / reinstalled `image-webpack-loader` or tried doing `npm rebuild` or the like, it just wasn't working. The only way was for me to delete my `/node_modules` folder and delete the `package-lock.json` and do a fresh `npm install` which built the module correctly for the current verison of Node I'm using.

The problem first cropped up for me because I changed Node.js version and I'm guessing the binaries were not built correctly for that version.

Take away: On changing Node.js versions, save yourself the trouble and just delete `/node_modules` and `package-json.lock` and do a fresh `npm install`.
